cinn/hlir/framework/graph_compiler.cc

```diff
@@ -1420,23 +1420,32 @@ std::vector<ir::LoweredFunc> GetFuncFromImpl(const std::shared_ptr<OpImpl>& impl
   // 4. Optimize the LoweredFunc
   VLOG(3) << "expr_pack.size() is : " << expr_pack.size();
   std::vector<ir::LoweredFunc> res;
-  for (int i = 0; i < expr_pack.size(); i++) {
-    if (funcs.size() > expr_pack.size()) {
-      auto new_args  = lang::GetArgs(funcs[i]->body, input_output_nodes);
-      funcs[i]->args = new_args;
+  CHECK_GE(funcs.size(), expr_pack.size());
+  if (funcs.size() > expr_pack.size()) {
+    for (int i = 0; i < funcs.size(); i++) {
+      for (int j = 0; j < expr_pack.size(); j++) {
+        Expr temp = expr_pack[j];
+        if (temp == funcs[i]->body) {
+          auto new_args  = lang::GetArgs(funcs[i]->body, input_output_nodes);
+          funcs[i]->args = new_args;
+          res.push_back(funcs[i]);
+        }
+      }
     }
-    auto temp_buffers   = lang::GetTempBuffers(all_arg_tensors, stages, funcs[i]->body);
-    funcs[i]->temp_bufs = temp_buffers;
-    funcs[i]->PrepareBufferCastExprs();
-    res.push_back(funcs[i]);
+  } else {
```

Comment:
结合前面`CHECK_GE`的判断，我觉得这儿还是明确写成：
```suggestion
  } else if (funcs.size() == expr_pack.size()) {
     res = funcs;
   } else {
     LOG(FATAL) << "The number of funcs should not less than expr_pack's";
   }
```

Reply:
Done, thanks